streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -349,6 +360,119 @@ public String queryableStoreName() {
         return toStream().selectKey(mapper);
     }
 
+    @Override
+    public KTable<K, V> suppress(final Suppress<K, V> suppress) {
+        final SuppressImpl<K, V> suppressImpl = (SuppressImpl<K, V>) suppress;
+        final String name = builder.newProcessorName(SUPPRESS_NAME);
+
+        final ProcessorSupplier<K, Change<V>> suppressionSupplier =
+            () -> new KTableSuppressProcessor<>(buildSuppress(suppressImpl));
+
+        final ProcessorParameters<K, Change<V>> processorParameters = new ProcessorParameters<>(
+            suppressionSupplier,
+            name
+        );
+
+        final ProcessorGraphNode<K, Change<V>> node = new ProcessorGraphNode<>(name, processorParameters, false);
+
+        builder.addGraphNode(this.streamsGraphNode, node);
+
+        return new KTableImpl<K, S, V>(
+            builder,
+            name,
+            suppressionSupplier,
+            keySerde,
+            valSerde,
+            Collections.singleton(this.name),
+            null,
+            false,
+            node
+        );
+    }
+
+    @SuppressWarnings("unchecked") // these casts are silly
+    private SuppressImpl<K, V> buildSuppress(final SuppressImpl<K, V> suppress) {
```

Comment:
Upon second look, I think I'll move these into a utility class to not pollute KTableImpl

Reply:
The basic idea is to traverse back through the topology until we find the window spec and get the grace period from it.